hyperspy/tests/test_events.py

```diff
@@ -0,0 +1,91 @@
+import nose.tools
+import hyperspy.events as he
+
+class TriggeredCorrectly: pass
+
+class TestEventsSuppression:
+        
+    def on_trigger(self, should_trigger):
+        if should_trigger:
+            raise TriggeredCorrectly()
+        else:
+            nose.tools.assert_true(False, "This event should not trigger")
+            
+    def trigger_check(self, trigger, should_trigger):
+        if should_trigger:
+            nose.tools.assert_raises(TriggeredCorrectly, trigger, True)
+        else:
+            trigger(False)
+        
+    def setUp(self):
+        self.events = he.Events()
+        
+    def reset(self):
```

Comment:
Oh, I thought it just called it once for the entire class. Please just
change it as necessary.
On Jan 16, 2015 7:00 PM, "Francisco de la Peña" <notifications@github.com>
wrote:

> In hyperspy/tests/test_events.py
> <https://github.com/francisco-dlp/hyperspy/pull/2#discussion-diff-23097074>
> :
>
> > +    def on_trigger(self, should_trigger):
> > +        if should_trigger:
> > +            raise TriggeredCorrectly()
> > +        else:
> > +            nose.tools.assert_true(False, "This event should not trigger")
> > +
> > +    def trigger_check(self, trigger, should_trigger):
> > +        if should_trigger:
> > +            nose.tools.assert_raises(TriggeredCorrectly, trigger, True)
> > +        else:
> > +            trigger(False)
> > +
> > +    def setUp(self):
> > +        self.events = he.Events()
> > +
> > +    def reset(self):
>
> I you call this method setUp nose will call it for your every time that
> it runs a test so that you don't need to call it manually.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/francisco-dlp/hyperspy/pull/2/files#r23097074>.
>